monitoredresource/gcp_metadata_config.go

```diff
@@ -70,13 +76,33 @@ func retrieveGCPMetadata() *gcpMetadata {
 	logError(err)
 	gcpMetadata.clusterName = strings.TrimSpace(clusterName)
 
+	clusterLocation, err := metadata.InstanceAttributeValue("cluster-location")
+	logError(err)
+
 	// Following attributes are derived from environment variables. They are configured
 	// via yaml file. For details refer to:
 	// https://cloud.google.com/kubernetes-engine/docs/tutorials/custom-metrics-autoscaling#exporting_metrics_from_the_application
 	gcpMetadata.namespaceID = os.Getenv("NAMESPACE")
 	gcpMetadata.containerName = os.Getenv("CONTAINER_NAME")
 	gcpMetadata.podID = os.Getenv("HOSTNAME")
 
+	// Monitoring API version can be obtained from cluster info.q
+	if gcpMetadata.clusterName != "" {
+		ctx := context.Background()
+		c, err := container.NewClusterManagerClient(ctx)
+		logError(err)
+
+		req := &containerpb.GetClusterRequest{
+			Name: fmt.Sprintf("projects/%s/locations/%s/clusters/%s", gcpMetadata.projectID, strings.TrimSpace(clusterLocation), gcpMetadata.clusterName),
+		}
+		resp, err := c.GetCluster(ctx, req)
```

Comment:
c could be nil.

Reply:
https://github.com/google/go-genproto/blob/master/googleapis/container/v1/cluster_service.pb.go#L1621

yes, but the check is there, that is why I ignored nil check, but to be safe I'll add this check